### PR TITLE
Implement Status column

### PR DESCRIPTION
**Pull request by KevinFCormier**
https://github.com/open-cluster-management/backlog/issues/14595

Table before and after:
image

Details before and after:
image

### CODE REVIEW

**Review by JustinKuli — COMMENTED** [review]
I didn't take a deep look, but nothing stands out as a potential issue.

One question: is Status just enabled/disabled? If someone asked me "what is the status of this policy" I would more likely have said if it's compliant or non-compliant. This might be more of a design thing, but have we considered combining the violation and status columns? In the user story, the design shows that if the policy is disabled, then the violations column should be empty.

Just thoughts, certainly not necessary to change anything at this point.

**Comment by KevinFCormier** [comment]
@JustinKuli Yeah, that's an interesting point on the term **Status**. I am not sure if @heyethankim considered a different name, or having it show for **Cluster violations** instead. I think it is good to keep it as a separate column for sorting and filtering purposes.

In the table view, for a disabled policy, it shows `--` for **Cluster violations**. In the details view, it is currently showing the icon and saying the status is unavailable for some clusters. This seems to be a pre-existing bug, but I will try to sort that out before merging.

**Comment by KevinFCormier** [comment]
Cluster violations fixed on details page:

image

File: src-web/tableDefinitions/grcPoliciesViewDef.js [hunk 1]
```diff
@@ -122,7 +138,7 @@ export default {
     },
   ],
```
Comment (dhaiducek) [inline src-web/tableDefinitions/grcPoliciesViewDef.js:139]: Now that "Disabled" is surfaced as a "Status" column, I think this `disabled` metadata can be removed.

File: src-web/tableDefinitions/utils.js [hunk 2]
```diff
@@ -136,6 +135,9 @@ function pushRows(items, rows, def, locale) {
           title: refetch ? key.transformFunction(item, locale, refetch) : key.transformFunction(item, locale),
           rawData: value
         }
+        if (key.textFunction && typeof key.textFunction === 'function') {
+          value.text = key.textFunction(item, locale)
+        }
```
Comment (dhaiducek) [inline src-web/tableDefinitions/utils.js:140]: Nice. I originally tried to capture strings via the `rawData` key, but adding a `textFunction` is probably clearer and more extensible.

**Review by dhaiducek — COMMENTED** [review]
Good work! This looks good to me.

File: src-web/tableDefinitions/grcPoliciesViewDef.js (hunk 1 above)
Comment (KevinFCormier) [reply src-web/tableDefinitions/grcPoliciesViewDef.js:139]: Ah yes. Thanks!

**Comment by heyethankim** [comment]
> @JustinKuli Yeah, that's an interesting point on the term **Status**. I am not sure if @heyethankim considered a different name, or having it show for **Cluster violations** instead. I think it is good to keep it as a separate column for sorting and filtering purposes.
> 
> In the table view, for a disabled policy, it shows `--` for **Cluster violations**. In the details view, it is currently showing the icon and saying the status is unavailable for some clusters. This seems to be a pre-existing bug, but I will try to sort that out before merging.

**I brought up this issue on the GRC channel. Mikela suggested that we should use "Cluster." But the issue is not that simple because we are using the term "Status" loosely.  We have at least 3 different instances. Let me finalize this before we implement https://github.com/open-cluster-management/backlog/issues/14596.**

image

**Review by fxiang1 — APPROVED** [review]
LGTM!

**Pull request merged**